src/puppet-web/firer.ts

```diff
@@ -315,22 +324,35 @@ async function checkRoomLeave(m: Message): Promise<void> {
   }
   /**
    * FIXME: leaver maybe is a list
+   * @lijiarui: I have checked, leaver will never be a list. If the bot remove 2 leavers at the same time, it will be 2 sys message, instead of 1 sys message contains 2 leavers.
    */
-  let leaverContact = room.member(leaver)
-
-  if (!leaverContact) {
-    log.error('PuppetWebFirer', 'fireRoomLeave() leaver %s not found, event `room-leave` & `leave` will not be fired', leaver)
-    return
+  let leaverContact: Contact | null, removerContact: Contact | null
+  if (leaver === this.userId) {
+    leaverContact = Contact.load(this.userId)
+    removerContact = room.member({alias: remover}) || room.member({name: remover})
```

Comment:
Please use the formal query filter name, instead of using an alias.

```ts
removerContact = room.member({roomAlias: remover}) || room.member({name: remover})
```

I will more prefer to write this piece of code in this way:
```ts
removerContact = room.member(remover)
```

Because `Room.member()` should do the job for translate the string name to a Contact inside a room.